### DISCUSSION TITLE
Welcome to Diomede Discussions!

### DISCUSSION THREAD

**Discussion by pradeeban** [KathiraveluLab/Diomede#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kolawolecreates** [comment]
Hello everyone,

I am Kolawole and I am interested in the tele health project.

I come from a part of the world that has accessibility issues such as Alaska that is being cited. I have a lot of passion for healthcare and excited to be a part of this project. 

I have reviewed documents on L4S and I have a solid background with networks protocols(TCP, UDP, QUIC), scheduling algorithms (FIFO, Robin Round, Priority scheduling etc.), congestion control, traffic management and further advanced network related concepts, I also have a diploma in Computer Networks to prove that. 

I am also familiar with Linux as an OS(Ubuntu and Kali Linux) and installing Linux servers. 

I would like to be a part of the project and take part in the challenges that gives me to be a part of the project.

Thank you.

**Reply by pradeeban** [reply]
Hi @kolawolecreates, welcome, and thank you for your interest.

We have submitted a project proposal to Google Summer of Code and await their announcements on accepted organizations: (February 21 - 18:00 UTC)

Hopefully, we will be accepted. Fingers crossed.

In any case, we warmly welcome contributions and collaborations.

**Reply by kolawolecreates** [reply]
Thank you for the response. Fingers crossed.

Any challenge or work that can be done in the meantime while we wait?

**Reply by pradeeban** [reply]
Getting to know L4S (such as configuring two nodes with them and sending data between them) is a good start. Other starting point is from DICOM endpoints, if we limit our focus to teleradiology (a realistic goal). Let me know when you have questions or updates in the process.

**Reply by kolawolecreates** [reply]
Thank you.

I would jump on that straight away and give updates as it comes.

**Reply by kolawolecreates** [reply]
Hello,

I have spent the past few weeks reviewing articles on L4S and I think I understand the principle. I have also done some practice with DICOM endpoints.


However, I am having difficulties with configuring two nodes and sending info between them. Am I to install multiple kernels on my Linux VM? Does the L4S have enough for me to install the node? Do I need physical layer devices?

**Reply by pradeeban** [reply]
Hi @kolawolecreates,
Nice to hear your updates.

The easier approach would be to have two linux machines in a local network. First confirm they could communicate locally over the Internet. Then, patch them with L4S/Linux distros.

So, you have L4S <--> L4S.

If you have 4 machines, it could be even better.

I am sure you could replicate the same with 4 VMs and a VM network. But showing anything realistic on network parameters, such as latency, on such virtual environment might be tricky.

I am of course aware that it is not like we have 2 - 4 computers lying around for us to patch them with L4S. But that will make an interesting and realistic observational study (rather than virtualization).

**Reply by kolawolecreates** [reply]
Errrm,

That makes more sense. No wonder I struggled.

I have just a Mac and a Windows. I use the Windows for Linux VM and scared to use the Mac for virtualisation because the M-series struggles with virtualisation.

Thinking of an alternative,

Would it work to setup let's say two VMs on Azure the free three and simulate from there?

**Reply by pradeeban** [reply]
Yes, Cloud VMs are a good option. We usually avoid those because they cost money (unless you have cloud credits or can use free tier for this).

**Comment by kolawolecreates** [comment]
Hello,

Thank you that is fine.

So the workflow would be to setup two linux VMs on Azure, install Linux kernel tree with L4S patches on both nodes and then try to send something from one to the other?

That should work for me?

**Reply by pradeeban** [reply]
That should work. I have never used Azure. Assuming you are comfortable using that. I have usually tested similar works with AWS EC2 VMs.

**Reply by kolawolecreates** [reply]
I have done that now and setup VMs on Azure and they were communicating with themselves. I tried to install L4S nodes with the steps on the readme on this page https://github.com/L4STeam/linux. I am hitting a brick wall when I run "sudo modprobe sch_dualpi2" it keeps showing "module not found" therefore testing can't go on.


What can I do?

**Reply by pradeeban** [reply]
Did you ask https://github.com/L4STeam/linux/issues or https://github.com/olgaalb/sch_dualpi2/issues ?

sch_dualpi2 does not seem to be an active repository.

**Reply by kolawolecreates** [reply]
Let me do so, thank you.

**Reply by kolawolecreates** [reply]
Yay application got accepted for Alaska

**Reply by pradeeban** [reply]
Thanks, don't forget to introduce yourself in our discussion forum. :)
We are excited.

**Reply by kolawolecreates** [reply]
I have done so. Trying to get my testing to work now.

**Reply by kolawolecreates** [reply]
Hello,

I have been able to setup the L4S nodes and send packets from one to the other on Azure

**Reply by pradeeban** [reply]
wow, this is excellent progress, @kolawolecreates! Now, you should be able to verify the claims of low latency and scalable throughput through some of the sample data that you can find on the L4S website.

**Reply by kolawolecreates** [reply]
Hello @pradeeban 

I have been observing and taking note of the claims of low latency and throughput. It is really cool. What next can I do?

**Reply by pradeeban** [reply]
@kolawolecreates:  I think you could work on the proposal draft. Make sure to clearly identify your findings and progress so far (since the evaluation is global across the organization and your proposal should stand out for the entire organization).

Feel free to use a new GitHub discussion to draft your proposal, or draft it elsewhere and share it with me.

**Reply by kolawolecreates** [reply]
Hello @pradeeban , I tried to start a discussion and that didn't seem to catch your attention. Can I send my draft over e-mail?

**Reply by pradeeban** [reply]
Yes, pls. Traveling these days until tomorrow. So, I was slow on catching up some notifications. Make sure to submit your proposal on the GSoC site on time as well.

**Comment by mmaurii** [comment]
Hi everyone;

My name is Maurizio. I discovered your post on the GSOC website. I found your purpose of research, like healthcare and climate science interesting, that really impacted my country. I’m studying IT at the University of Bologna (UNIBO) and I have a passion for trekking and cycling.

I already have some skill in java and a little bit in python, but I have also seen other languages like C#, html, javascript, typescript, php. I’m now in the middle of my first degree and I’m really interested in discovering the world of open source projects, which is why I’m here.

I would like to be a part of the project. In particular I’m interested in project number [4]. I’m now looking at DICOM, Kheops and MEDIator. If anyone has any advice on what else to watch or do, it’ s welcome.

**Comment by Shimul-Baidya** [comment]
Hi there👋,

My name is Shimul Baidya, and I am a second-year Computer Science & Engineering student at Jahangirnagar University, Bangladesh. I am an enthusiast about open source and eager to become an active member of the community.

I am really interested in one of the projects, "Creating shareable 'albums' from locally stored DICOM images" as part of Google Summer of Code.

Why am I interested?
I want to be an active member of the open source community from now. And I would love to start my journey with GSoC as a contributor. I also aspire to be mentor in future. Also I appreciate the purpose of research that is being done by this community.

My qualifications:
I am quite comfortable with python and java programming language. I am also familiar with technologies like linux, C/C++.

Currently I am learning about DICOM, MEDlator, Kheops and other terminologies mentioned in the project. Help of any kind would really be appreciated to know about the project better.

Therefore, I am very interested to contribute to this project and collaborate with other members of the community. Hopefully, we will learn a lot of things together. Thank you.

**Comment by manvendra5345** [comment]
Hi everyone!
I am Manvendra Singh Solanki from Cse Ai-Ml background and i am also interested in working in this project , I know python , java , HTML , CSS  . I can adapt very fast and have great learning capacity but these thing are common what i think i have different in me is that i want to learn from thoes who are better than me and catch up  to them and even beat them and always keep learning.

**Comment by EbrahemMordy** [comment]
Hi everyone!

I am Ebrahem Mordy from Egypt, I am a final year CS student and also a competitive programmer and a region finalist in the Arab And African regions. I am also an Expert in codeforces [profile](https://codeforces.com/profile/M0rdy) and now I am trying to be SWE, especially a java backend dev using spring boot

I aim to enter gsoc this year and make it my year and when checking organizations find this one and I liked this project (Creating shareable "albums" from locally stored DICOM images) I see it's okay for me as I can use Java so I want to know if there is any knowledge I must learn to understand this project and the final output for it, also if there is any issue I can check and solve to start contributing in this organization.

**Comment by M-Haseeb01** [comment]
I’m Muhammad Haseeb , a 3rd year Computer Science student with an interest in open-source development, machine learning, and web applications. I’m excited to be part of this community and contribute to the DICOM album creation project for Google Summer of Code.

I have experience in Python, Django, SQL, and Machine Learning, and I’m eager to learn more about DICOM images and medical imaging frameworks. Looking forward to collaborating, learning, and making meaningful contributions!
